kobuki_auto_docking/CMakeLists.txt

```diff
@@ -1,43 +1,81 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(kobuki_auto_docking)
-find_package(catkin REQUIRED COMPONENTS roscpp rospy nodelet pluginlib message_filters actionlib kdl_conversions
-                                        ecl_threads ecl_geometry ecl_linear_algebra kobuki_msgs kobuki_dock_drive nav_msgs)
-
-catkin_package(
-   INCLUDE_DIRS include
-   LIBRARIES kobuki_auto_docking_ros kobuki_auto_docking_nodelet
-   CATKIN_DEPENDS roscpp rospy nodelet pluginlib message_filters actionlib kdl_conversions
-                  ecl_threads ecl_geometry ecl_linear_algebra kobuki_msgs kobuki_dock_drive nav_msgs
-)
 
-include_directories(include
-                    ${catkin_INCLUDE_DIRS})
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-add_library(kobuki_auto_docking_ros src/auto_docking_ros.cpp)
-add_library(kobuki_auto_docking_nodelet src/nodelet.cpp)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-add_dependencies(kobuki_auto_docking_ros ${catkin_EXPORTED_TARGETS})
+find_package(ament_cmake_ros REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclpy REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(rclcpp_action REQUIRED)
+find_package(rcl_interfaces REQUIRED)
 
-target_link_libraries(kobuki_auto_docking_ros ${catkin_LIBRARIES})
-target_link_libraries(kobuki_auto_docking_nodelet ${catkin_LIBRARIES} kobuki_auto_docking_ros)
+find_package(kobuki_ros_interfaces REQUIRED)
+find_package(kobuki_core REQUIRED)
 
-install(TARGETS kobuki_auto_docking_ros kobuki_auto_docking_nodelet
-        DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-)
-install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+find_package(message_filters REQUIRED)
+find_package(tf2_ros REQUIRED)
+
+find_package(ecl_geometry REQUIRED)
+find_package(ecl_linear_algebra REQUIRED)
+
+find_package(nav_msgs REQUIRED)
```

Comment:
For this whole set of `find_package` calls, please just put them all together and alphabetize them.  It will make it easier to find.  Also I think you need to `find_package(std_msgs REQUIRED)`, `find_package(tf2_geometry_msgs REQUIRED)`, and `find_package(geometry_msgs REQUIRED)` in here, and add them as dependencies of the `kobuki_auto_docking` library.